src/Features/CSharp/Portable/CodeRefactorings/ConvertIfToSwitch/CSharpConvertIfToSwitchCodeRefactoringProvider.Pattern.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertIfToSwitch
+{
+    partial class CSharpConvertIfToSwitchCodeRefactoringProvider
+    {
+        private abstract class Pattern : IPattern
+        {
+            SyntaxNode IPattern.CreateSwitchLabel() => CreateSwitchLabel();
+
+            protected abstract CasePatternSwitchLabelSyntax CreateSwitchLabel();
```

Comment:
> can we not have `Pattern<T>`

I think I'm mistaken. Currently there is no `Pattern<T>` and we can't have in it in the base either unless we add a type parameter for `Analyzer` for that. is that what you meant?